source/render/RenderManager.cpp

```diff
@@ -665,6 +666,19 @@ void RenderManager::rrTemporalMarkTile(Tile* curTile)
     ent->setVisible(bb);
 }
 
+void RenderManager::rrDetachEntity(GameEntity* curEntity)
+{
+    Ogre::SceneNode* node = mSceneManager->getSceneNode(curEntity->getOgreNamePrefix() + curEntity->getName() + "_node");
```

Comment:
You have GameEntity::getEntityNode that would avoid to search the whole scene. AFAIK, it is set for all game entities.